packages/openapi/src/decorators.ts

```diff
@@ -225,64 +214,24 @@ function omitUndefined<T extends Record<string, unknown>>(data: T): T {
   return Object.fromEntries(Object.entries(data).filter(([k, v]) => v !== undefined)) as any;
 }
 
-function validateIsUri(context: DecoratorContext, url: string, propertyName: string) {
-  try {
-    new URL(url);
-    return true;
-  } catch {
-    reportDiagnostic(context.program, {
-      code: "not-url",
-      target: context.getArgumentTarget(0)!,
-      format: { property: propertyName, value: url },
-    });
-    return false;
-  }
-}
-
-function validateAdditionalInfoModel(context: DecoratorContext, typespecType: TypeSpecValue) {
+function validateAdditionalInfoModel(
+  context: DecoratorContext,
+  typespecType: TypeSpecValue,
+  data: AdditionalInfo & Record<`x-${string}`, unknown>,
+) {
   const propertyModel = context.program.resolveTypeReference(
     "TypeSpec.OpenAPI.AdditionalInfo",
   )[0]! as Model;
-
+  const diagnostics: Diagnostic[] = [];
   if (typeof typespecType === "object" && propertyModel) {
-    const diagnostics = checkNoAdditionalProperties(
-      typespecType,
-      context.getArgumentTarget(0)!,
-      propertyModel,
+    diagnostics.push(
```

Comment:
just report, no need to push to an array?

Reply:
it need report all error/warning once. so push array first, then call `context.program.reportDiagnostics(diagnostics);`